acl.go

```diff
@@ -300,43 +301,44 @@ func (odbi *ovndb) rowToACL(uuid string) *ACL {
 
 // Get all acl by lswitch
 func (odbi *ovndb) aclListImp(lsw string) ([]*ACL, error) {
-	var listACL []*ACL
-
 	odbi.cachemutex.RLock()
 	defer odbi.cachemutex.RUnlock()
 
 	cacheLogicalSwitch, ok := odbi.cache[TableLogicalSwitch]
 	if !ok {
 		return nil, ErrorNotFound
 	}
-	var lsFound bool
 	for _, drows := range cacheLogicalSwitch {
 		if rlsw, ok := drows.Fields["name"].(string); ok && rlsw == lsw {
 			acls := drows.Fields["acls"]
 			if acls != nil {
 				switch acls.(type) {
 				case libovsdb.OvsSet:
 					if as, ok := acls.(libovsdb.OvsSet); ok {
-						for _, a := range as.GoSet {
+						listACL := make([]*ACL, len(as.GoSet))
```

Comment:
Thanks for optimization and improvements.

Instead of  manually tracking a new index i, better set capacity for slice as length is known which will save unnecessary memory allocations too
e.g. listACL := make([]*ACL, 0, len(as.GoSet)). 

keep the listACL = append(listACL, ta) as is so that it will add from start of slice.

Reply:
@noah8713 Thanks. I thought append() function always reallocate memory for the new result, apprently it also reuse the memory as long as the capacity is sufficient. I will update my code and repush. Thanks again.